app/Http/Controllers/CategoriesController.php

```diff
@@ -0,0 +1,77 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use App\Http\Requests\ValidateCategory;
+use App\Models\Categories;
+use Illuminate\Http\Request;
+use Illuminate\Support\Str;
+
+class CategoriesController extends Controller
+{
+    public function index()
+    {
+        $categories = Categories::all();
+
+        return view('sections.categories.categories-page', ['categories' => $categories]);
+    }
+
+    public function createCategory()
+    {
+        return view('sections.categories.create-category');
+    }
+
+    public function store(ValidateCategory $request)
+    {
+        $request->validated();
+
+        $category = new Categories();
+        $category->name = $request->name;
+        $category->description = $request->description;
+        $category->slug = Str::slug($request->name);
+        $category->save();
+
+        return redirect()->route('categories.categories')->with('success', 'Categoria criada com successo');
+    }
+
+    public function show($id)
+    {
+        $category = Categories::find($id);
+
+        return view('sections.categories.edit-category', ['category' => $category]);
+    }
+
+    public function update(Request $request, $id)
+    {
+        $category = Categories::find($id);
```

Comment:
Quando tu faz um `find`, esse método pode retornar `null` (se tu ver a doc do método essa informação está lá no retorno). 
Sendo assim se o usuário passar na url por exemplo `admin/categories/6323` e não existir uma categoria de id `6323`, tua aplicação vai quebrar pois logo a frente quando tu acessar `$category->name` a `category` será `null` assim gerando um erro 500 na aplicação.

Nesse teu caso pra ti não precisar manualmente verificar se `category` é null, tu pode fazer o seguinte:
```suggestion
        $category = Categories::query()->findOrFail($id);
```

Esse metodo tenta achar uma categoria, e se não achar ele retorna um erro 404 (Not foud) para o front.